### ISSUE TITLE
Significant speed decrease in training WAN 2.2 LORA since PR #585

### ISSUE THREAD

**Issue by Enyakk** [kohya-ss/musubi-tuner#632]
I am training a LORA for WAN 2.2 and noticed quite low training speed of 24sec/it. Switching back to the commit before merging https://github.com/kohya-ss/musubi-tuner/pull/585 has decreased training speed back down to 8sec/it again.

While the gains from the new block-swapping methode are significant and appreciated it seems like only having the new method could be a drawback for training speed in certain configurations.

training config for LORA:
- WAN 2.2 high-noise
- Rank 4 (very low)
- Block Swap 29
- mixed precision fp16
- batch size 4
- 512x512 image dataset

[config.txt](https://github.com/user-attachments/files/22710475/config.txt)

Pytorch = 2.8.0, CUDA 12, RTX 4090

**Comment by Enyakk** [comment]
I will also set up a completely fresh and factory-installed musubi-tuner repository in a bit to validate it is not due to incompatible libraries.

**Comment by Enyakk** [comment]
I retested with a fresh factory-clean installation of the current repository and PyTorch 2.5 and I am seeing the same performance degradation. So I am confident this is not due to an installation issue at least. It could be the training configuration or a OS level setting

**Comment by kohya-ss** [comment]
Thank you for reporting this. The new block swap method has a slow transfer speed per block, so it is possible that the speed will decrease if almost all blocks are swapped.

If we set batch size to 2 and gradient accumulation to 2, we can reduce the number of swapped blocks while maintaining roughly the same training settings. What will the overall speed be?

Edit: I will compare the old and new methods with the maximum transfer block size on Wan.

**Comment by Enyakk** [comment]
Thank you I will give your recommendation a try.

**Comment by elie-h** [comment]
@kohya-ss what about full finetunes? Also seem to be massively slowed down.
